Java/measureeval/src/main/java/com/lantanagroup/link/measureeval/FileSystemInvocation.java

```diff
@@ -0,0 +1,109 @@
+            } else {
+                throw new IllegalArgumentException("Unsupported measure bundle file format: " + measureBundlePath);
+            }
+        } else {
+            // Parse and load each file in the directory
+            Bundle bundle = new Bundle();
+            bundle.setType(Bundle.BundleType.COLLECTION);
+
+            File[] files = measureBundleFile.listFiles();
+
+            if (files != null) {
+                List<String> loaded = new ArrayList<>();
+
+                for (File file : files) {
+                    String filePath = file.getAbsolutePath();
+                    if (file.isFile() && (filePath.endsWith(".json") || filePath.endsWith(".xml"))) {
+                        String fileName = file.getName().substring(0, file.getName().lastIndexOf('.'));
+
+                        if (loaded.contains(fileName)) {
+                            System.out.println("Skipping duplicate file: " + filePath);
+                        }
+
+                        Resource resource;
+
+                        if (filePath.endsWith(".json")) {
+                            resource = (Resource) fhirContext.newJsonParser().parseResource(FileUtils.readFileToString(file));
+                        } else {
+                            resource = (Resource) fhirContext.newXmlParser().parseResource(FileUtils.readFileToString(file));
+                        }
+
+                        bundle.addEntry(new Bundle.BundleEntryComponent().setResource(resource));
+                        loaded.add(fileName);
+                    } else {
+                        System.out.println("Skipping file: " + filePath);
+                    }
+                }
+            }
+
+            System.out.println("Loaded " + bundle.getEntry().size() + " resources from directory: " + measureBundlePath);
+
+            return bundle;
+        }
+    }
+
+    public static void main(String[] args) {
+        String measureBundlePath = args[0];
+        String patientBundlePath = args[1];
+        String start = args[2];
+        String end = args[3];
+
+        try {
+            Bundle measureBundle = getBundle(measureBundlePath);
+            Bundle patientBundle = getBundle(patientBundlePath);
+            MeasureEvaluator evaluator = MeasureEvaluator.compile(fhirContext, measureBundle);
+            Patient patient = patientBundle.getEntry().stream()
+                    .filter(e -> e.getResource() instanceof Patient)
+                    .map(e -> (Patient) e.getResource())
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException("Patient resource not found in bundle"));
+            var report = evaluator.evaluate(new DateTimeType(start), new DateTimeType(end),
+                    new StringType("Patient/" + patient.getIdElement().getIdPart()), patientBundle);
+            String json = fhirContext.newJsonParser().encodeResourceToString(report);
+            System.out.println(json);
```

Comment:
Again, probably overkill, but could encode directly to `System.out` (by wrapping it in an `OutputStreamWriter`) with `encodeResourceToWriter`.

Reply:
Using logging system instead. May consider output parameter in future.